update/2_3.sql

```diff
@@ -721,8 +721,8 @@ UPDATE `zz_plugins` SET `name` = 'Serial' WHERE `name` = 'Lotti';
 UPDATE `zz_plugins` SET `title` = `name` WHERE `title` = '';
 
 -- Nuova struttura per i plugin Sedi e Referenti in Anagrafiche
-UPDATE `zz_plugins` SET `script` = '', `options` = '	{ "main_query": [	{	"type": "table", "fields": "Nome, Indirizzo, Città, CAP, Provincia, Referente",	"query": "SELECT an_sedi.id, an_sedi.nomesede AS Nome, an_sedi.indirizzo AS Indirizzo, an_sedi.citta AS Città, an_sedi.cap AS CAP, an_sedi.provincia AS Provincia, an_referenti.nome AS Referente FROM an_sedi LEFT OUTER JOIN an_referenti ON idsede = an_sedi.id WHERE 1=1 AND an_sedi.idanagrafica=|idanagrafica| HAVING 2=2 ORDER BY an_sedi.id DESC"}	]}', `directory` = 'sedi', `version` = '2.3', `compatibility` = '2.*' WHERE `name` = 'Sedi';
-UPDATE `zz_plugins` SET `script` = '', `options` = '	{ "main_query": [	{	"type": "table", "fields": "Nominativo, Mansione, Telefono, Indirizzo email, Sede",	"query": "SELECT an_referenti.id, an_referenti.nome AS Nominativo, mansione AS Mansione, an_referenti.telefono AS Telefono, an_referenti.email AS \'Indirizzo email\', an_sedi.nomesede AS Sede FROM an_referenti LEFT OUTER JOIN an_sedi ON idsede = an_sedi.id WHERE 1=1 AND an_referenti.idanagrafica=|idanagrafica| HAVING 2=2 ORDER BY an_referenti.id DESC"}	]}', `directory` = 'referenti', `version` = '2.3', `compatibility` = '2.*' WHERE `name` = 'Referenti';
+UPDATE `zz_plugins` SET `script` = '', `options` = '    { "main_query": [   {   "type": "table", "fields": "Nome, Indirizzo, Città, CAP, Provincia, Referente", "query": "SELECT an_sedi.id, an_sedi.nomesede AS Nome, an_sedi.indirizzo AS Indirizzo, an_sedi.citta AS Città, an_sedi.cap AS CAP, an_sedi.provincia AS Provincia, an_referenti.nome AS Referente , an_anagrafiche.idanagrafica FROM an_sedi LEFT OUTER JOIN an_referenti ON idsede = an_sedi.id LEFT OUTER JOIN an_anagrafiche ON an_anagrafiche.idanagrafica = an_sedi.idanagrafica WHERE 1=1 AND an_sedi.idanagrafica=|idanagrafica| HAVING 2=2 ORDER BY an_sedi.id DESC"} ]}', `directory` = 'sedi', `version` = '2.3', `compatibility` = '2.*' WHERE `name` = 'Sedi';
```

Comment:
L'unica differenza rispetto alla versione precedente è l'aggiunta di `an_anagrafiche.idanagrafica` nel SELECT e il relativo JOIN.

Il JOIN mancante era è effettivamente la causa del problema indicato nella issue #55, ma il SELECT di `an_anagrafiche.idanagrafica`  non è necessario e quindi rimovibile; lo stesso vale per la riga successiva.

Potresti inoltre descrivere a livello generico la risoluzione di un determinato problema nella campo descrizione del commit?

Reply:
@lepool Certamente, ma l'utilizzo di un campo all'interno di un WHERE non dipende dalla sua presenza nel SELECT ma da quella "virtuale" nel FROM.
In pratica, attraverso il FROM con il nuovo JOIN il WHERE può utilizzare tutti i nuovi campi creati dal JOIN senza averli obbligatoriamente nel SELECT; per l'**HAVING** la presenza nel SELECT è obbligatoria.